tests/integration/targets/vm_snapshot/tasks/01_vm_snapshot_info.yml

```diff
@@ -0,0 +1,172 @@
+
+    # --------------------------------------------------------
+
+    # ======== Test with no parameters =========
+    - name: List all VM snapshots
+      scale_computing.hypercore.vm_snapshot_info:
+      register: vm_snapshots
+    - ansible.builtin.debug:
+        var: vm_snapshots.records
+    - ansible.builtin.assert: &assert-existing
+        that:
+          - vm_snapshots is not changed
+          - vm_snapshots.records != []
+          - vm_snapshots.records[0].keys() | sort ==
+            ["automated_trigger_timestamp",
+            "block_count_diff_from_serial_number",
+            "domain",
+            "domain_uuid",
+            "label",
+            "local_retain_until_timestamp",
+            "remote_retain_until_timestamp",
+            "replication",
+            "timestamp",
+            "type",
+            "vm_uuid"]
+    - ansible.builtin.assert:
+        that:
+          - vm_snapshots.records[0].label == existing_snapshot_label
+
+     # ====== Test every parameter one by one =======
+    - name: List all VM snapshots with snapshot label "{{ non_unique_snapshot_label }}"
+      scale_computing.hypercore.vm_snapshot_info:
+        label: "{{ non_unique_snapshot_label }}"
+      register: vm_snapshots
+    - ansible.builtin.debug:
+        var: vm_snapshots.records
+    - ansible.builtin.assert: *assert-existing
+    - ansible.builtin.assert:
+        that:
+          - vm_snapshots.records[0].label == "{{ non_unique_snapshot_label }}"
+
+    - name: List all VM snapshots with a nonexistent label
+      scale_computing.hypercore.vm_snapshot_info:
+        label: "{{ nonexistent_snapshot_label }}"
+      register: vm_snapshots
+    - ansible.builtin.debug:
+        var: vm_snapshots.records
+    - ansible.builtin.assert: &assert-nonexistent
+        that:
+          - vm_snapshots is not changed
+          - vm_snapshots.records == []
+
+    - name: List all VM snapshots with vm_name "snapshot-test-vm-1"
+      scale_computing.hypercore.vm_snapshot_info:
+        vm_name: "snapshot-test-vm-1"
+      register: vm_snapshots
+    - ansible.builtin.debug:
+        var: vm_snapshots.records | length
+    - ansible.builtin.assert: *assert-existing
+    - ansible.builtin.assert:
+        that:
+          - vm_snapshots.records | length == 3  # there are 3 snapshots of vm snapshot-test-vm-1
+          - vm_snapshots.records[0].domain.name == "snapshot-test-vm-1"
+
+    - name: List all VM snapshots with a nonexistent vm_name
+      scale_computing.hypercore.vm_snapshot_info:
+        vm_name: "{{ nonexistent_snapshot_vm_name }}"
+      register: vm_snapshots
+    - ansible.builtin.debug:
+        var: vm_snapshots.records
+    - ansible.builtin.assert: *assert-nonexistent
+
+    - name: List all VM snapshots with serial "3"  # the last snapshot serial of each vm is 3, so there should be 2 snapshots with the serial 3
+      scale_computing.hypercore.vm_snapshot_info:
+        serial: 3
```

Comment:
If you have only `serial: 3`, then you might get snapshots from other unrelated VMs. So `vm_snapshots.records | length >= 2` is more reliable test.

But you can add extra check - are the two expected VM names in the returned list. Something like
`vm1_uuid in vm_snapshots.records | map(attribute='vm_uuid')`.